src/v/cloud_storage_clients/client.h

```diff
@@ -19,6 +19,7 @@
 #include <seastar/core/lowres_clock.hh>
```

Comment:
>cloud_storage_clients/{client,s3_client}: added client::delete_object…
>…s. this interface is low level

please limit the length of the subject line in a git commit message so that it doesn't wrap. 72 characters is safe.